src/table.js

```diff
@@ -488,30 +489,44 @@ Table.prototype.createReadStream = function(options) {
   if (options.limit) {
     reqOpts.rowsLimit = options.limit;
   }
-
+  const chunkFormatter = new ChunkFormatter();
   var stream = pumpify.obj([
     this.requestStream(grpcOpts, reqOpts),
-    through.obj(function(data, enc, next) {
-      var throughStream = this;
-      var rows = Row.formatChunks_(data.chunks, {
-        decode: options.decode,
-      });
-
-      rows.forEach(function(rowData) {
-        if (stream._ended) {
-          return;
+    through.obj(
+      function(data, enc, next) {
+        var throughStream = this;
+        try {
+          chunkFormatter.formatChunks(
+            data.chunks,
+            {
+              decode: options.decode,
+            },
+            (err, rowData) => {
+              if (stream._ended) {
+                return false;
+              } else {
+                var row = self.row(rowData.key);
+                row.data = rowData.data;
+                throughStream.push(row);
+                return true;
```

Comment:
No return necessary. This whole block could look like:

```js
if (stream._ended) {
  return;
}

var row = self.row(rowData.key);
row.data = rowData.data;
throughStream.push(row);
```

Reply:
After further digging, I see how the `return` value is used as the meaning of `shouldContinue` in ChunkFormatter. Can we be more direct about that, and just have a method like `chunkFormatter.exit()` (or `stopProcessing()`, etc). It's not clear that this function has any duties that work backwards into ChunkFormatter, and it would be easy to imagine a contributor or maintainer in the future getting rid of the `true`/`false` aspect in a refactor.